pubnub-core/src/pubnub.rs

```diff
@@ -164,18 +130,17 @@ where
     /// message; the other 2 streams will be blocked waiting for that 30-second duration on the
     /// slow consumer.
     ///
-    /// To workaround this problem, you may consider enabling reduced resiliency with
-    /// [`PubNubBuilder::reduced_resliency`], which will drop messages on the slowest consumers,
-    /// allowing faster consumers to continue processing messages without blocking.
```

Comment:
I agree that backpressure is a valuable feature to support. There are a number of approaches we could use there. I'd like to approach it separately as a dedicated PR, to focus on properly organizing not just the interfaces, but also testing capabilities so that we can validate our backpressure works as intended.
The logic around backpressure is often very tricky, and if not tested rarely correct on the first attempt. :)

Let's leave this for now, and revisit it sometime later.

Reply:
Design-wise, a similar approach is offered by the https://docs.rs/futures/0.3.4/futures/sink/trait.Sink.html

Technically we can just switch subscribe loop to operate via `Sink`s, and we'll naturally have to handle backpressure within the loop.

Reply:
I guess I should do some research on `Sink`... I'm not sure what else there is to do. `futures_channel::mpsc::Sender` already implements `Sink`. Which aligns with my thinking that we would just switch the `send` to `try_send`, so we can handle back pressure without blocking the entire task.

But happy to hear other thoughts!

Reply:
I'd say the main difficulty there is not how to detect that sink is stuck - but to actually implement the logic described at the comment for the now removed option:
https://github.com/pubnub/rust/blob/821f3eb7ffd1fa42c3fc830f81097dbeb4b9966f/pubnub-core/src/pubnub.rs#L423-L443
Anyhow, I think we should create a dedicated issue and more the discussion there - that would be a more appropriate place.

Reply:
Created #32